providers/github/resources/github.go

```diff
@@ -4,11 +4,44 @@
 package resources
 
 import (
+	"context"
 	"time"
 
 	"github.com/google/go-github/v57/github"
+	"github.com/kofalt/go-memoize"
```

Comment:
if that code is not updated and third pary dependencies are not updated often, we may need to handle this as a util internally.